### PR TITLE
IoW-13 Hugo Page

### PR DESCRIPTION
**Pull request by abriggs-usgs**
Before making a pull request
----------------------------

- [x] Update the changelog appropriately

IOW-13 Hugo Page
-----------
Created basic Hugo generated landing page for labs.waterdata.usgs.gov (URL is still in the process of creation, and not yet active). 

Description
-----------
- Created a Docker container to run Hugo locally and for use by a Jenkins' job to create the static website that can be deployed to Amazon Web Services (AWS)
- Created a Jenkins job to build and then deploy static site to AWS. https://jenkins.wma.usgs.chs.ead/view/WDFN/job/waterdata_build%20_deploy_pipeline/

-Created three tiers of AWS S3 buckets
 AWS https://s3.console.aws.amazon.com/s3/home?region=us-east-2
www.labs-development.waterdata.usgs.gov (development)
www.labs-staging.waterdata.usgs.gov (staging)
www.labs.waterdata.usgs.gov (production)

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

File: .idea/libraries/R_User_Library.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<component name="libraryTable">
```
Comment (duselmann) [inline .idea/libraries/R_User_Library.xml]: you might want to ignore .idea directory

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      agent {
+        dockerfile {
+          additionalBuildArgs "--build-arg HUGO_COMMAND=build --build-arg DEPLOY_TIER=$DEPLOY_TIER --build-arg AWS_SYNC=sync_yes"
+          args '-u root:root -v "${WORKSPACE}":/src'
+          reuseNode true
+        }
+      }
+      steps {
+        echo 'Docker image build complete, preparing to run image/container/hugo'
+        sh "/src/buildDeploy.sh build ${DEPLOY_TIER} sync_yes"
```
Comment (duselmann) [inline Jenkinsfile]: aha, we discussed something like this. Nice work finding what we needed.

File: README.md [hunk 3]
```diff
@@ -1 +1,62 @@
...
+```
+
+Disclaimer
+----------
+This software is preliminary or provisional and is subject to revision. It is being provided to meet the need for
+timely best science. The software has not received final approval by the U.S. Geological Survey (USGS). No warranty,
+expressed or implied, is made by the USGS or the U.S. Government as to the functionality of the software and
+related material nor shall the fact of release constitute any such warranty. The software is provided on the
+condition that neither the USGS nor the U.S. Government shall be held liable for any damages resulting from
+the authorized or unauthorized use of the software.
```
Comment (duselmann) [inline README.md]: should we add pipeline documentation?

File: awsTierDeploy.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+URL=$1
+
+aws s3 sync /src/public s3://${URL} --delete
+echo "Ran AWS sync command."
```
Comment (duselmann) [inline awsTierDeploy.sh]: we discussed this together on Monday. I like the echo note.

File: .idea/libraries/R_User_Library.xml (hunk 1 above)
Comment (abriggs-usgs) [reply .idea/libraries/R_User_Library.xml]: Will do.

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+version: "3"
+
+services:
+  hugo:
+    build: .
+    command: server --buildDrafts
+    ports:
+      - "1313:1313"
+    volumes:
+      - .:/src
```
Comment (duselmann) [inline docker-compose.yml]: nice and short and sweet.

File: hugoCommand.sh [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        ;;
+    server)
+        cd /src/themes/wdfn_theme
+        npm install
+        npm run watch & cd /src && hugo server --bind=0.0.0.0 ${ARGS}
+        ;;
+    *)
+    exec "$@"
+        ;;
+esac
```
Comment (duselmann) [inline hugoCommand.sh]: I do not know Hugo but I saw this work in your local space on Monday.

File: themes/wdfn_theme/package.json [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    "postcss-flexbugs-fixes": "^4.1.0",
+    "rollup": "^0.66.2",
+    "rollup-plugin-alias": "^1.4.0",
+    "rollup-plugin-buble": "^0.19.2",
+    "rollup-plugin-commonjs": "^9.1.8",
+    "rollup-plugin-node-resolve": "^3.4.0",
+    "rollup-plugin-uglify": "^6.0.0",
+    "uglify-js": "^3.4.9"
+  }
+}
```
Comment (duselmann) [inline themes/wdfn_theme/package.json]: has anyone looked at the version updates for these dependencies? (should we?)

**Comment by duselmann** [comment]
I primarily reviewed the components related to the story to get jenkins to run the dockerfile. I only glanced at the content because I presumed it must have been review as it was created in other stories.

File: .idea/libraries/R_User_Library.xml (hunk 1 above)
Comment (duselmann) [reply .idea/libraries/R_User_Library.xml]: After the dir is added to .gitignore the files have to be deleted as a followup commit.

File: themes/wdfn_theme/package.json (hunk 7 above)
Comment (abriggs-usgs) [reply themes/wdfn_theme/package.json]: Yeah, I was going to look into getting Dependabot running and scanning the dependencies. I'm sure some of them could use updating.

File: README.md (hunk 3 above)
Comment (abriggs-usgs) [reply README.md]: Probably . . . but I feel really shaky on the how the pipelines work, and am worried that what I end up saying will prove misleading.

File: .idea/libraries/R_User_Library.xml (hunk 1 above)
Comment (abriggs-usgs) [reply .idea/libraries/R_User_Library.xml]: Ha. Working on it. I always forget how to get the files out of Git once it is added.

**Pull request merged**